### ISSUE TITLE
window does not re-render under WebGpu, only renders during (un)focus event

### ISSUE THREAD

**Issue by jokeyrhyme** [wez/wezterm#3384]
### What Operating System(s) are you seeing this problem on?

Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

```
nu ❯ paru -Qi sway
Name            : sway
Version         : 1:1.8.1-1

nu ❯ paru -Qi wlroots
Name            : wlroots
Version         : 0.16.2-1

nu ❯ paru -Qi mesa
Name            : mesa
Version         : 23.0.0-3

nu ❯ paru -Qi amdvlk
Name            : amdvlk
Version         : 2022.Q4.4-3

nu ❯ paru -Qi vulkan-radeon
Name            : vulkan-radeon
Version         : 23.0.0-3
```

### WezTerm version

wezterm 20230326-195053-5b511792 (nightly appimage)

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

I don't understand the cause (yet), but for me, the new WebGpu renderer doesn't seem to update the contents of the window

I have to change focus with other windows to trigger a re-render in `wezterm` when running WebGpu

- screen capture of me typing in the new wezterm window and nothing appearing: https://youtu.be/C0HH6T-USc0

- screen capture of me typing in the new wezterm window and changing focus to get the new contents to appear: https://youtu.be/dFE8noeDBT8

This happens for me on 2 different systems with the same software setup/versions:
- AMD Ryzen 7950X + Radeon 6800XT (with integrated GPU disabled and not visible in `enumerate_gpus()`)
- Dell Precision 5550 (with nVidia GPU disabled and not visible in `enumerate_gpus()`)

### To Reproduce

1. run `wezterm --config 'front_end='WebGpu"'`
2. in the new terminal window that appears, type something
3. observe that you can't see what you just typed
4. hit Enter
5. observe that you can't see the results of the command that you just invisibly ran
6. focus a different window and focus back
7. observe that you can now see what you previously typed and ran

### Configuration

I've reproduced this with wezterm.lua containing just

```
return {}
```

### Expected Behavior

- in theory, (and for probably the majority of other wezterm users) the terminal window should show the updated contents the same way it does when not using WebGpu

### Logs

```
Debug Overlay
wezterm version: 20230326-195053-5b511792 x86_64-unknown-linux-gnu
Window Environment: Wayland
OpenGL version: WebGPU: name=AMD Radeon RX 6800 XT (RADV NAVI21), device_type=DiscreteGpu, backend=Vulkan, driver=radv, driver_info=Mesa 23.0.0, vendor=4098, device=29631
Enter lua statements or expressions and hit Enter.
Press ESC or CTRL-D to exit
```

```
> wezterm.gui.enumerate_gpus()
[
    {
        "backend": "Vulkan",
        "device": 29631,
        "device_type": "DiscreteGpu",
        "driver": "radv",
        "driver_info": "Mesa 23.0.0",
        "name": "AMD Radeon RX 6800 XT (RADV NAVI21)",
        "vendor": 4098,
    },
    {
        "backend": "Gl",
        "device": 0,
        "device_type": "Other",
        "name": "AMD Radeon RX 6800 XT (navi21, LLVM 15.0.7, DRM 3.49, 6.2.8-arch1-1)",
        "vendor": 4098,
    },
]
```

(even the Debug Overlay doesn't render unless I change focus to a different window)

**Comment by jdelkins** [comment]
ACK for me on hyprland and sway using amdvlk. Gl driver works fine.

```
Debug Overlay
wezterm version: 20230320-124340-559cb7b0 x86_64-unknown-linux-gnu
Window Environment: Wayland
OpenGL version: WebGPU: name=AMD Radeon RX 480 Graphics, device_type=DiscreteGpu, backend=Vulkan, driver=AMD open-source driver, driver_info=2022.Q4.4 (LLPC), vendor=4098, device=26591
Enter lua statements or expressions and hit Enter.
Press ESC or CTRL-D to exit
11:56:43.736 WARN wezterm_gui::termwindow::resize > cannot resize window to match Some(RowsAndCols { rows: 24, cols: 80 }) because window_state is MAXIMIZED
> wezterm.gui.enumerate_gpus()
[
    {
        "backend": "Vulkan",
        "device": 26591,
        "device_type": "DiscreteGpu",
        "driver": "AMD open-source driver",
        "driver_info": "2022.Q4.4 (LLPC)",
        "name": "AMD Radeon RX 480 Graphics",
        "vendor": 4098,
    },
    {
        "backend": "Gl",
        "device": 0,
        "device_type": "Other",
        "name": "AMD Radeon RX 480 Graphics (polaris10, LLVM 15.0.7, DRM 3.49, 6.2.8-zen1-1-zen)",
        "vendor": 4098,
    },
]
```

**Comment by jokeyrhyme** [comment]
This might be related to or even a duplicate of #3126

**Comment by jokeyrhyme** [comment]
Closed, duplicate of #3126
